mautic/mautic
# Added relative date support to anniversary/birthday

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL |  https://github.com/mautic/documentation/pull/299
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This update bring support relative dates to anniversary.  
At the moment (https://www.mautic.org/docs/en/contacts/manage_segments.html) we can use anniversary for date filter. This update allow use anniversary with relative dates. For example:

- anniversary -7 days
- anniversary -1 month
- anniversary +3 days
- birthday +1 month
- birthday -14 days

etc.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com/6993)
2. Create date contact custom field
3. Setup this field with date to anniversary with -7 days  
For example: If today is 11. 12. 2018 setup this field for 4. 12. 2010
4. Create segment with this custom field filter and add equals  and value of filter: **anniversary -7 days**
5. Your contact should be added to segment
6. If you change contacts custom field  value to 3. 12. 2010, contact should be removed from segment.
